ShopProjectHomeWork/ShopProjectHomeWork/SearchScreen/GoodsViewController.swift

```diff
@@ -0,0 +1,299 @@
+        goodsNameSmallLabel.textAlignment = .center
+        view.addSubview(goodsNameSmallLabel)
+    }
+
+    private func addGoodsLigthGrayButton() {
+        goodsLigthGrayButton = UIButton(frame: CGRect(
+            x: view.center.x - 40,
+            y: goodsNameSmallLabel.frame.origin.y + goodsNameSmallLabel.frame.height + 40,
+            width: 35,
+            height: 35))
+        goodsLigthGrayButton.clipsToBounds = true
+        goodsLigthGrayButton.layer.cornerRadius = 0.5 * goodsLigthGrayButton.bounds.size.width
+        goodsLigthGrayButton.backgroundColor = .lightGray
+        view.addSubview(goodsLigthGrayButton)
+    }
+
+    private func addGoodsDarkGrayButton() {
+        goodsDarkGrayButton = UIButton(frame: CGRect(
+            x: view.center.x + 5,
+            y: goodsNameSmallLabel.frame.origin.y + goodsNameSmallLabel.frame.height + 40,
+            width: 35,
+            height: 35))
+        goodsDarkGrayButton.clipsToBounds = true
+        goodsDarkGrayButton.layer.cornerRadius = 0.5 * goodsDarkGrayButton.bounds.size.width
+        let gradientLayer = CAGradientLayer()
+        gradientLayer.frame = goodsDarkGrayButton.bounds
+        gradientLayer.colors = [#colorLiteral(red: 0.194206357, green: 0.1725068986, blue: 0.1928621531, alpha: 1).cgColor, #colorLiteral(red: 0.3491136748, green: 0.3297520554, blue: 0.34427327, alpha: 1).cgColor]
+
+        goodsDarkGrayButton.layer.shadowColor = UIColor.yellow.cgColor
+        goodsDarkGrayButton.layer.shadowOpacity = 1
+        goodsDarkGrayButton.layer.shadowOffset = .zero
+        goodsDarkGrayButton.layer.shadowRadius = 10
+
+        goodsDarkGrayButton.layer.insertSublayer(gradientLayer, at: 0)
+        view.addSubview(goodsDarkGrayButton)
+    }
+
+    private func addGreenOkImageView() {
+        greenOkImageView = UIImageView(frame: CGRect(
+            x: 90,
+            y: goodsDarkGrayButton.frame.origin.y + goodsDarkGrayButton.frame.height + 30,
+            width: 25,
+            height: 25))
+        greenOkImageView.image = UIImage(systemName: "checkmark.circle")
+        greenOkImageView.clipsToBounds = true
+        greenOkImageView.layer.cornerRadius = 12
+        greenOkImageView.layer.borderWidth = 5
+        greenOkImageView.layer.borderColor = UIColor.black.cgColor
+        greenOkImageView.backgroundColor = #colorLiteral(red: 0.2592384219, green: 0.824129045, blue: 0.3576895595, alpha: 1)
+        greenOkImageView.tintColor = .black
+        view.addSubview(greenOkImageView)
+    }
+
+    private func addGoodsCompatibleLabel() {
+        goodsCompatibleLabel = UILabel(frame: CGRect(
+            x: greenOkImageView.frame.origin.x + greenOkImageView.frame.width + 10,
+            y: goodsDarkGrayButton.frame.origin.y + goodsDarkGrayButton.frame.height + 32,
+            width: view.frame.width - 20,
+            height: 20))
+        let string = Strings.goodsCompatibleLabelText
+        let rangeOne = (string as NSString).range(of: "Совместимо с ")
```

Comment:
```suggestion
        let rangeOne = (string as NSString).range(of: " ")
```
все текстовки в enum Constants